examples/shop-page/Components/ProductItem.lua

```diff
@@ -57,13 +46,37 @@ function ProductItem:render()
 		PriceLabel = Roact.createElement("TextLabel", {
 			AnchorPoint = Vector2.new(0.5, 1),
 			BackgroundTransparency = 1,
+			Font = Enum.Font.SourceSans,
 			Text = ("R$ %d"):format(props.price),
 			TextColor3 = Color3.fromRGB(10, 200, 10),
 			TextScaled = true,
+			TextStrokeTransparency = 0,
+			TextStrokeColor3 = Color3.fromRGB(255, 255, 255),
 			Position = UDim2.new(0.5, 0, 1, 0),
 			Size = UDim2.new(1, 0, 0.3, 0),
 		}),
 	})
 end
 
+function ProductItem.validateProps(props)
+	return pcall(function()
+		assert(
+			type(props.image) == "string",
+			("props.image should be a string (got %q)"):format(type(props.image))
+		)
+		assert(
+			type(props.price) == "number",
+			("props.price should be a number (got %q)"):format(type(props.price))
+		)
+		assert(
+			type(props.productId) == "string" or type(props.productId) == "number",
+			("props.productId should be a string or a number (got %q)"):format(type(props.productId))
+		)
+		assert(
+			props.order == nil or type(props.order) == "number",
+			("props.order should be a number (got %q)"):format(type(props.order))
+		)
+	end)
+end
```

Comment:
We should move this to the top of the file!

Now that I'm looking at it in writing (without having `t`), I'm way less enthusiastic about typechecking here. :(

The motivation of my original comment about having types in a comment is that it's generally a poor practice that has caused rot in code internally. It's easy for those comments to get out of sync with the real expectations of the component, but much harder for `validateProps` to!

Reply:
Yes I agree that comments get easily out of sync, and I did not know about `validateProps` so thanks for getting me up to speed on that!

So you would not go with type checking in validate props? What would you propose in that case? 

I could add a value constraint on the price to be positive.

Reply:
I'm not sure. I kinda wish we could just depend on `t` here.